indexer/cmd/main.go

```diff
@@ -295,6 +321,11 @@ func runMigrations(lggr logger.Logger, dbURI, migrationsDir string) error {
 		return err
 	}
 
+	err = ensureDBConnection(lggr, db)
+	if err != nil {
+		return fmt.Errorf("could not connect to database: %w", err)
+	}
```

Comment:
Some basic reconnect logic - in the K8s deployments it should be able to handle these kinds of situations (restarts due to the application shutting down with a non-zero error code), however restart policies set in testcontainers don't seem to be properly working.